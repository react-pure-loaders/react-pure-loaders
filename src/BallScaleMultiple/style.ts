import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import delay from '../utilities';
import { Grid } from '../types';

const size = '60px';

const ballScaleMultiple = keyframes({
  '0%': {
    opacity: 0,
    transform: 'scale(0.0)',
  },
  '100%': {
    opacity: 0,
    transform: 'scale(1.0)',
  },
  '5%': {
    opacity: 1,
  },
});

export function ballScale() {
  const n = 3;
  let i = 2;
  const grid: Grid = {};
  for (; i < 4; i += 1) {
    grid[`> div:nth-of-type(${i})`] = {
      animationDelay: delay('0.2s', n, i),
    };
  }

  return grid;
}

export default (color?: string) => ({
  ...ballScale(),
  position: 'relative' as 'relative',
  transform: `translateY(-${size} / 2)`,
  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    animation: `${ballScaleMultiple} 1s 0s linear infinite`,
    height: size,
    left: '-30px',
    margin: '0',
    opacity: 0,
    position: 'absolute' as 'absolute',
    top: '0px',
    width: size,
  },
});
