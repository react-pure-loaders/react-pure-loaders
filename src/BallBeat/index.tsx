/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
