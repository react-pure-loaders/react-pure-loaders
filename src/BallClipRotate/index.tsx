/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
