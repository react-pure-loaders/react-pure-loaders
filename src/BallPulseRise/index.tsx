/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
