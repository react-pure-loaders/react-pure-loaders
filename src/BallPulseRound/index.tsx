/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
