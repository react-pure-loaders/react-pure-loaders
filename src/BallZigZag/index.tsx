/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
