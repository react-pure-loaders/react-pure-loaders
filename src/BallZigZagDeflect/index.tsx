/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
