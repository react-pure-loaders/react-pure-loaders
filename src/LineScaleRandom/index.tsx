/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
