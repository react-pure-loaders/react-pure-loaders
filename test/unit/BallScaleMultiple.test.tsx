import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallScaleMultiple from '../../src/BallScaleMultiple';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallScaleMultiple>', () => {
    afterEach(cleanup);

    test('BallScaleMultiple should match snapshot', () => {
        const { container } = render(<BallScaleMultiple loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallScaleMultiple should have default color', () => {
        const { container } = render(<BallScaleMultiple loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test('BallScaleMultiple should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallScaleMultiple color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallScaleMultiple should have no children', () => {
        const { container } = render(<BallScaleMultiple loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallScaleMultiple should have three children', () => {
        const { container } = render(<BallScaleMultiple loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(4);
    });
});
