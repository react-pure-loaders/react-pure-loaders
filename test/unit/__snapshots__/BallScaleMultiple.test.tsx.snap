// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallScaleMultiple> BallScaleMultiple should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0;
    -webkit-transform: scale(0.0);
    -moz-transform: scale(0.0);
    -ms-transform: scale(0.0);
    transform: scale(0.0);
  }

  100% {
    opacity: 0;
    -webkit-transform: scale(1.0);
    -moz-transform: scale(1.0);
    -ms-transform: scale(1.0);
    transform: scale(1.0);
  }

  5% {
    opacity: 1;
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translateY(-60px / 2);
  -moz-transform: translateY(-60px / 2);
  -ms-transform: translateY(-60px / 2);
  transform: translateY(-60px / 2);
}

.emotion-0>div:nth-of-type(2) {
  -webkit-animation-delay: -0.20s;
  animation-delay: -0.20s;
}

.emotion-0>div:nth-of-type(3) {
  -webkit-animation-delay: 0.00s;
  animation-delay: 0.00s;
}

.emotion-0>div {
  border-radius: 100%;
  height: 60px;
  margin: 0;
  width: 60px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1s 0s linear infinite;
  animation: animation-0 1s 0s linear infinite;
  left: -30px;
  opacity: 0;
  position: absolute;
  top: 0px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
  <div />
</div>
`;
